### PR TITLE
WIP: Upgrade PkgDev to a new 1.0 era

### LINKED ISSUE #143: ERROR: LoadError: UndefVarError: Dir not defined with Julia 0.7
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0 (2018-08-08 06:46 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> using Pkg

julia> Pkg.add("PkgDev")
  Updating registry at `C:\Users\scls\.julia\registries\General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
 Installed PkgDev ─ v0.2.1
  Updating `C:\Users\scls\.julia\environments\v0.7\Project.toml`
  [149e707d] + PkgDev v0.2.1
  Updating `C:\Users\scls\.julia\environments\v0.7\Manifest.toml`
  [682c06a0] + JSON v0.19.0
  [4d1e1d77] + Nullables v0.0.7
  [149e707d] + PkgDev v0.2.1

julia> using PkgDev
[ Info: Precompiling PkgDev [149e707d-584d-56d3-88ec-740c18e106ff]
┌ Warning: Deprecated syntax `try without catch or finally` at C:\Users\scls\.julia\packages\PkgDev\DJ4wQ\src\generate.jl:127.
└ @ C:\Users\scls\.julia\packages\PkgDev\DJ4wQ\src\generate.jl:127
ERROR: LoadError: UndefVarError: Dir not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at .\sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at .\boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at .\loading.jl:1038
 [5] include(::Module, ::String) at .\sysimg.jl:29
 [6] top-level scope at none:2
 [7] eval at .\boot.jl:319 [inlined]
 [8] eval(::Expr) at .\client.jl:399
 [9] top-level scope at .\none:3
in expression starting at C:\Users\scls\.julia\packages\PkgDev\DJ4wQ\src\PkgDev.jl:15
ERROR: Failed to precompile PkgDev [149e707d-584d-56d3-88ec-740c18e106ff] to C:\Users\scls\.julia\compiled\v0.7\PkgDev\KD6BZ.ji.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] macro expansion at .\logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at .\loading.jl:1185
 [4] _require(::Base.PkgId) at .\logging.jl:311
 [5] require(::Base.PkgId) at .\loading.jl:852
 [6] macro expansion at .\logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at .\loading.jl:834
```

If `PkgDev` shouldn't be used with Julia 0.7 and above, maybe this repository should be deprecated and a link to a document how packages should be created be displayed.

I'm aware of https://docs.julialang.org/en/latest/stdlib/Pkg/#Creating-your-own-packages-1

I liked `PkgDev` much than having to switch between:
- Shell console
- Julia REPL-mode
- Pkg REPL-mode 
That's just my opinion... but Pkg REPL-console probably have much feature!

Kind regards

### LINKED ISSUE #145: Include `using Pkg` in the travis scripts

### PR DESCRIPTION
**Pull request by KristofferC**
Very WIP!

Fixes #145 
Fixes #143

### CODE REVIEW

**Comment by KristofferC** [comment]
TODO:

- [ ] Update commit message (include diff link)
- [ ] Consistency check for registry
- [ ] Update the PR code (perhaps use GitHub.jl)
- [ ] Create a toy registry and ask people to try it out.

**Comment by Nosferican** [comment]
What's the status on this?

**Comment by dpo** [comment]
Bump. Currently BinaryBuilder doesn't build on 0.7 because of #143, which this PR should (?) fix.

**Comment by Nosferican** [comment]
Any updates?

**Comment by mkborregaard** [comment]
When is this planned?

**Comment by davidanthoff** [comment]
Bump, would be awesome to have this :)

File: README.md [hunk 1]
```diff
@@ -1,54 +1,69 @@
-# Julia Package Development Kit (PDK)
+# Julia Package Development Kit
 
 [![Build Status](https://travis-ci.org/JuliaLang/PkgDev.jl.svg?branch=master)](https://travis-ci.org/JuliaLang/PkgDev.jl)[![Build status](https://ci.appveyor.com/api/projects/status/gnd6dqbdaxcx1c23/branch/master?svg=true)](https://ci.appveyor.com/project/wildart/pkgdev-jl/branch/master)
 
-PkgDev.jl provides a set of tools for a developer to create, maintain and register packages in Julia package repository, a.k.a. [METADATA](https://github.com/JuliaLang/METADATA.jl).
+PkgDev.jl provides a set of tools for a developer to create, maintain and register packages in a Julia package registry, for example (but not limited to) [METADATA](https://github.com/JuliaLang/METADATA.jl).
```
Comment (Nosferican) [inline README.md]: Change this to General by now?

**Comment by KristofferC** [comment]
https://github.com/JuliaComputing/Registrator.jl is the future.

**Comment by mkborregaard** [comment]
But - does that also create a local package with the correct folder/file structure? Or is PkgTemplates the official recommendation for this?

**Comment by KristofferC** [comment]
Yeah, seems good for us to just make PkgTemplates the goto place for this.

**Comment by adamslc** [comment]
For what its worth, I don't think that `Registrator` can completely replace the `PkgDev` workflow. Here is my use case: I maintain a small handful of private packages, and I maintain a private registry for these packages. I cannot publish the packages, even to a private GitHub repo. Currently, I manually update the registry, but I had hoped that I would eventually be able to switch to a more automated process. I strongly believe that I'm not the only one with these sorts of requirements.

Would there be opposition to me reviving this PR to try to get `PkgDev` into a usable state with the `Pkg3` ecosystem?

**Pull request closed without merging**